chips/sam4l/src/pm.rs

```diff
@@ -283,18 +304,23 @@ unsafe fn configure_48mhz_dfll() {
 }
 
 /// Configure the system clock to use the 16 MHz external crystal directly
-unsafe fn configure_external_oscillator(startup_mode: OscStartupMode) {
+unsafe fn configure_external_oscillator(frequency: OscillatorFrequency,
+                                        startup_mode: OscillatorStartup) {
     // Use the cache
     flashcalw::FLASH_CONTROLLER.enable_cache();
 
     // Need the 32k RC oscillator for things like BPM module and AST.
     bscif::enable_rc32k();
 
     // start the external oscillator
-    match startup_mode {
-        OscStartupMode::FastStart => scif::setup_osc_16mhz_fast_startup(),
-        OscStartupMode::SlowStart => scif::setup_osc_16mhz_slow_startup(),
-    };
+    match frequency {
+        OscillatorFrequency::Frequency16MHz => {
+            match startup_mode {
+                OscillatorStartup::FastStart => scif::setup_osc_16mhz_fast_startup(),
+                OscillatorStartup::SlowStart => scif::setup_osc_16mhz_slow_startup(),
+            };
```

Comment:
Making a match like this means any future modifications will be required by the compiler to add code and make sure they are handled correctly.